### PR TITLE
Convert format and lint scripts to null safety.

### PR DESCRIPTION
**Pull request by gspencergoog**
This converts the `ci/bin/format.dart` and `ci/bin/lint.dart` scripts to null safety.

The only package they depend on that isn't yet null safe is the `isolate` package.  Once that is converted, they can run with sound null safety.

### CODE REVIEW

File: ci/bin/format.dart [hunk 1]
```diff
@@ -106,17 +104,17 @@ Future<String> _runGit(
...
 /// Base class for format checkers.
 ///
 /// Provides services that all format checkers need.
 abstract class FormatChecker {
   FormatChecker({
-    ProcessManager /*?*/ processManager,
-    @required this.baseGitRef,
-    @required this.repoDir,
-    @required this.srcDir,
+    ProcessManager? processManager,
```
Comment (cbracken) [inline ci/bin/format.dart:114]: Could we default here instead of in the initialiser list?

File: ci/bin/format.dart [hunk 2]
```diff
@@ -231,7 +225,7 @@ abstract class FormatChecker {
           'failed to apply.');
       completedJobs
           .where((WorkerJob job) => job.result.exitCode != 0)
-          .map<String>((WorkerJob job) => job.result.output)
+          .map<String?>((WorkerJob job) => job.result.output)
```
Comment (cbracken) [inline ci/bin/format.dart:228]: It looks like output is never null:
https://pub.dev/documentation/process_runner/latest/process_runner/ProcessRunnerResult/output.html

File: ci/bin/format.dart [hunk 3]
```diff
@@ -334,7 +328,7 @@ class ClangFormatChecker extends FormatChecker {
     );
   }
 
-  /*late*/ File clangFormat;
+  late File clangFormat;
```
Comment (cbracken) [inline ci/bin/format.dart:331]: This looks like it could be `late final`, or maybe even `static final` since it's always the same thing and statics are lazy in Dart.

File: ci/bin/format.dart [hunk 4]
```diff
@@ -459,7 +453,7 @@ class JavaFormatChecker extends FormatChecker {
     );
   }
 
-  /*late*/ File googleJavaFormatJar;
+  late File googleJavaFormatJar;
```
Comment (cbracken) [inline ci/bin/format.dart:456]: This also looks like it could be `late final` or `static final`.

File: ci/bin/format.dart [hunk 5]
```diff
@@ -599,7 +593,7 @@ class GnFormatChecker extends FormatChecker {
     );
   }
 
-  /*late*/ File gnBinary;
+  late File gnBinary;
```
Comment (cbracken) [inline ci/bin/format.dart:596]: One more `late final` or `static final`.

File: ci/bin/format.dart [hunk 6]
```diff
@@ -711,7 +705,7 @@ class WhitespaceFormatChecker extends FormatChecker {
     return true;
   }
 
-  static Future<_GrepResult> _hasTrailingWhitespace(File file) async {
+  static Future<_GrepResult?> _hasTrailingWhitespace(File file) async {
```
Comment (cbracken) [inline ci/bin/format.dart:708]: What do you think about a `_GrepResult.empty(File file)` constructor that generates a result with empty hits/lineNumbers lists? Could be a separate patch if you prefer to focus primarily on the types here.

File: ci/bin/format.dart [hunk 7]
```diff
@@ -861,7 +855,7 @@ Future<int> main(List<String> arguments) async {
           'On Windows, only whitespace and gn checks are currently supported.');
   parser.addFlag('verbose', help: 'Print verbose output.', defaultsTo: verbose);
 
-  ArgResults options;
+  late ArgResults options;
```
Comment (cbracken) [inline ci/bin/format.dart:858]: `late final`?

**Review by cbracken — APPROVED** [review]
This made my evening! So much nicer. lgtm modulo nits. Apologies if I'm mistaken about some of the `late final` or `static final`s in there.

LGTM stamp from a Japanese personal seal

File: ci/bin/format.dart (hunk 1 above)
Comment (gspencergoog) [reply ci/bin/format.dart:114]: Yes, that is more idiomatic. I did it the other way before because passing on an argument you received from another constructor might send in an unexpected explicit null. But with null safety we don't need to do that anymore! Yay!

**Comment by chinmaygarde** [comment]
The issue with the [presubmits should have been fixed](https://chat.google.com/room/AAAAm69vf-M/qetJIggG2KQ). Please rebase this and lets land this ASAP.

**Comment by chinmaygarde** [comment]
We weren't sure if you needed to rebase TBH. In any case, I re-ran the test and am adding the CQ label. If it fails, please rebase.
